### ISSUE TITLE
Do you have a plan to release your train code?

### ISSUE THREAD

**Issue by IrvingShu** [almazan/deep-image-retrieval#4]

**Comment by fereenwong** [comment]
Waiting for train code +1

**Comment by almazan** [comment]
Hi @IrvingShu and @fereenwong,

Unfortunately, we don't have plans for releasing the training code at the moment.

However, I just added to the repository the AP loss function that is described in the paper and that we used to train our models, which can be used for reference. You can find it [here](https://github.com/almazan/deep-image-retrieval/blob/master/dirtorch/loss.py).

I hope that helps :)

Cheers!
Jon

**Comment by yan796** [comment]
hi, thanks for the code :). I was wondering if label should not be NxN ? Can you please help understand what label is from your equation (12) ? Is it Yi ?

**Comment by lyl472324464** [comment]
Why do we need loss function during reference? :o

**Comment by lyl472324464** [comment]
> Hi @IrvingShu and @fereenwong,
> 
> Unfortunately, we don't have plans for releasing the training code at the moment.
> 
> However, I just added to the repository the AP loss function that is described in the paper and that we used to train our models, which can be used for reference. You can find it [here](https://github.com/almazan/deep-image-retrieval/blob/master/dirtorch/loss.py).
> 
> I hope that helps :)
> 
> Cheers!
> Jon

Why do we need loss function during reference? :o

**Comment by mrgransky** [comment]
@lyl472324464 for training process

**Comment by lyl472324464** [comment]
> @lyl472324464 for training process

Got it.Thank you

**Comment by liyongqing-1101** [comment]
@yan796 @almazan Could you please simply explain why the shape of the label is N * M, why isn't N by N?

**Comment by Emalude** [comment]
> @yan796 @almazan Could you please simply explain why the shape of the label is N * M, why isn't N by N?

I actually have a similar question. If the descriptors are N * M (images * embed_dimension) shouldn't the labels be just N? Maybe I am confusing embedding and descriptors?
